ambry-commons/src/main/java/com.github.ambry.commons/HelixNotifier.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.commons;
+
+import com.github.ambry.config.HelixPropertyStoreConfig;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import org.apache.helix.AccessOption;
+import org.apache.helix.ZNRecord;
+import org.apache.helix.store.HelixPropertyListener;
+import org.apache.helix.store.HelixPropertyStore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * <p>
+ *    A {@link Notifier} implementation using {@link HelixPropertyStore} as the underlying communication mechanism.
+ * Each topic is created as a {@link ZNRecord} under the {@code topicPath}. A message for that topic will
+ * be written to the simple field of the {@link ZNRecord} in the form of {@code "message": "message content"}.
+ * Each {@link ZNRecord} under the {@code topicPath} will be used for a single topic.
+ * </p>
+ * <p>
+ *   If multiple {@link HelixNotifier} running on different node are configured with the same
+ *   {@link HelixPropertyStoreConfig}, they will essentially be in the same topic domain. I.e., {@link TopicListener}s
+ *   can subscribe through any of these {@link HelixNotifier}s for a topic. A message of that topic sent through
+ *   {@link HelixNotifier}-1 on node-1 can be received by all the {@link TopicListener}s on other nodes.
+ * </p>
+ */
+public class HelixNotifier implements Notifier<String> {
+  private static final String TOPIC_PATH = "/topics";
+  private static final String MESSAGE_KEY = "message";
+  private final Logger logger = LoggerFactory.getLogger(getClass());
+  private final HelixPropertyStore<ZNRecord> helixStore;
+  private final Map<TopicListener, HelixPropertyListener> topicListenerToHelixListenerMap = new ConcurrentHashMap<>();
+
+  /**
+   * Constructor.
+   * @param storeConfig The {@link HelixPropertyStoreConfig} that specifies necessary configs to start a HelixNotifier.
+   * @param storeFactory The factory to start a {@link HelixPropertyStore}.
+   */
+  public HelixNotifier(HelixPropertyStoreConfig storeConfig, HelixPropertyStoreFactory<ZNRecord> storeFactory) {
+    if (storeFactory == null) {
+      throw new IllegalArgumentException("storeFactory cannot be null.");
+    }
+    List<String> subscribedPaths = new ArrayList<>();
+    subscribedPaths.add(storeConfig.rootPath + TOPIC_PATH);
+    helixStore = storeFactory.getHelixPropertyStore(storeConfig, subscribedPaths);
```

Comment:
Can you see if `Collections.singletonList(storeConfig.rootPath)` works instead of creating an `ArrayList`?